pkg/schema/type.go

```diff
@@ -564,6 +576,55 @@ func (n *NullType) SetDeprecated(deprecated bool, notice string) {
 	n.documentation.deprecated = deprecated
 }
 
+// GetValidations provides validations from @schema/validation for a node
+func (t *DocumentType) GetValidations() []assertions.Rule {
+	if t.validations != nil {
+		return t.validations
+	}
+	return nil
```

Comment:
```suggestion
	return t.validations
```
if `t.validations` is _in fact_ `nil`, then `nil` will be returned.

(applies to the other getters, below)